app/presenters/licences/view-licence.presenter.js

```diff
@@ -46,6 +57,21 @@ function _generateLicenceHolder (licenceHolder) {
   return licenceHolder
 }
 
+function _generatePurposes (licenceVersions) {
+  if (!licenceVersions || licenceVersions.length === 0 || licenceVersions[0]?.purposes.length === 0) {
+    return null
+  }
+
+  const uniquePurposes = [...new Set(licenceVersions[0].purposes.map((item) => {
+    return item.description
+  }))]
```

Comment:
Just for clarity in my opinion I'd split this into separate statements.

```suggestion
  const allPurposes = licenceVersions[0].purposes.map((item) => {
    return item.description
  })
  
  const uniquePurposes = [...new Set(allPurposeDescriptions)]
```